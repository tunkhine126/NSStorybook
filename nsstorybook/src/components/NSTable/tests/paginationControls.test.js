/**
 *
 * Tests for PaginationControls
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import PaginationControls from '../paginationControls';

Enzyme.configure({ adapter: new Adapter() });

describe('<PaginationControls />', () => {
  const wrapper = mount(
    <TestingComponent>
      <PaginationControls
        count={20}
        page={1}
        rowsPerPage={5}
        resultCount={50}
        currentPage={1}
        totalPages={1}
        onChangePage={() => {}}
      />
    </TestingComponent>
  );

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <PaginationControls
          count={100}
          page={0}
          currentPage={1}
          totalPages={1}
          rowsPerPage={15}
          resultCount={50}
        />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Click the next arrow without errors', () => {
    const spy = jest.spyOn(global.console, 'error');
    const nextBtn = wrapper.find('button#next-btn').first();
    nextBtn.simulate('click');

    expect(spy).not.toHaveBeenCalled();
  });

  it('Click the prev arrow without errors', () => {
    const spy = jest.spyOn(global.console, 'error');
    const prevBtn = wrapper.find('button#prev-btn').first();
    prevBtn.simulate('click');

    expect(spy).not.toHaveBeenCalled();
  });

  it('Manually change the page from input without errors', () => {
    const spy = jest.spyOn(global.console, 'error');
    const inputBox = wrapper.find('input#page-input').first();

    inputBox.simulate('change', { target: { value: 2 } });
    inputBox.simulate('change', { target: { value: -1 } });

    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <PaginationControls
          count={100}
          page={0}
          currentPage={1}
          totalPages={1}
          rowsPerPage={15}
          resultCount={50}
        />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
