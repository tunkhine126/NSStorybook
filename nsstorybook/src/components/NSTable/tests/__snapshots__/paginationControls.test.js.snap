// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationControls /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between MuiGrid-grid-xs-10"
>
  <div
    class="MuiGrid-root makeStyles-numbeOfRequests-23 MuiGrid-item"
  >
    <p
      class="MuiTypography-root makeStyles-dataResults-10 MuiTypography-body2 MuiTypography-colorInherit"
    >
      50
       
      <span>
        of
      </span>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorInherit"
    >
      100
       
      <span>
        requests
      </span>
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-200 MuiButton-textPrimary makeStyles-textPrimary-293 Mui-disabled Mui-disabled"
      disabled=""
      id="prev-btn"
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label makeStyles-label-292"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-paginationLeftArrow-24 MuiSvgIcon-fontSizeInherit"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
          />
        </svg>
        <span>
          Prev
        </span>
      </span>
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormControl-root MuiTextField-root paginationPageInput MuiFormControl-marginNormal"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              aria-label="Page Number"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
              disabled=""
              id="page-input"
              maxlength="3"
              type="number"
              value="1"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-282 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-283"
                style="width: 0.01px;"
              >
                <span>
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <span>
          of
        </span>
         
        1
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-200 MuiButton-textPrimary makeStyles-textPrimary-295 Mui-disabled Mui-disabled"
      disabled=""
      id="next-btn"
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label makeStyles-label-294"
      >
        <span>
          Next
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-paginationRightArrow-25 MuiSvgIcon-fontSizeInherit"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
